lib/kubeclient/common.rb

```diff
@@ -218,7 +218,26 @@ def self.underscore_entity(entity_name)
 
     def create_rest_client(path = nil)
       path ||= @api_endpoint.path
+      RestClient::Resource.new(@api_endpoint.merge(path).to_s, common_options)
+    end
+
+    def do_rest_request(args)
+      uri = @api_endpoint.merge("#{@api_endpoint.path}/#{@api_version}/#{args[:path]}")
       options = {
+        url: uri.to_s,
+        method: args[:method],
+        headers: (args[:headers] || {}).merge('params' => args[:params])
+      }.merge(common_options)
+      options[:block_response] = args[:block] if args[:block]
+      resp = RestClient::Request.execute(options)
+      return unless resp.instance_of?(Net::HTTPClientError)
```

Comment:
Not returning the response?
Does Request.execute return a Net::HTTPClientError outside of block_response mode?
=> Both OK as it's the only use case for this method. I'd make it unconditionally expect a block.

[style: early return usually used for problems, with flat fallthrough to happy path.  doesn't matter.]

Reply:
@cben I haven't tested it but I expect that it would; I don't think `block_response` changes the behavior in this regard.

Reply:
@cben i originally used an `if` block to check for error and fallthrough on the happy path, and rubocop complained that i should replace the `if` block with a guard clause...